meta-mbl-apps/recipes-connectivity/mbl-cloud-client/mbl-cloud-client.bb

```diff
@@ -139,6 +140,8 @@ do_install() {
 
     install -d "${D}${systemd_unitdir}/system/"
     install -m 0644 "${WORKDIR}/mbl-cloud-client.service" "${D}${systemd_unitdir}/system/"
+
+    install -d "${D}${MBL_BOOTFLAGS_DIR}"
```

Comment:
Why do we need to add an empty directory in the package?

Reply:
So that it exists when the update scripts attempt to write flag files there.

We _could_ have the update scripts create the directory if it doesn't exist, but I don't really want to do that - I want them to be agnostic about how the directory is created (whether it's a simple `mkdir` or whether it requires a `mount`).

...Or was that a rhetorical question because you think I should add a code comment? I'll add a code comment.

Reply:
Well, I don't think that makes sense the recipe packing an empty directory. I think that the source code responsibility to handle the dependencies, in this case creating a directory, if they don't exist.

Reply:
> Well, I don't think that makes sense the recipe packing an empty directory.

Why?

For the reasons I stated above, I don't want the source code to create the directory.

Reply:
Comment added in https://github.com/ARMmbed/meta-mbl/pull/641/commits/474b0cdef7d933b77c7e70226c3ab77c52c4ff39.

Reply:
Right, having the knowledge of what kind of operation needs to be performed to have this path in place is a good reason for not embedding it into the code. But the same _problem_ applies at the recipe level. Correct?

Is this `MBL_BOOTFLAGS_DIR` supposed to be used by other recipes/packages/apps? If so, this path needs to be provided by a _common_ recipe like base-files.

Reply:
> But the same problem applies at the recipe level. Correct?

Sort-of, but less so. How the directory is created is platform and distro specific, and I think it's more reasonable to have that sort of thing in recipes than in source code.

> Is this MBL_BOOTFLAGS_DIR supposed to be used by other recipes/packages/apps?

Mmmm... Good point. I'll create it in base-files.

Reply:
Moved to base-files in https://github.com/ARMmbed/meta-mbl/pull/641/commits/9275a7d3a84479a25a49da5bd21058908df6fe6c and https://github.com/ARMmbed/meta-mbl/pull/641/commits/67fa872a925d91bc91ca2c56e817c9d2f3023052.